driver/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java

```diff
@@ -155,6 +157,43 @@ public int available() {
     }
 
     @Override
+    public void mark() {
+        mark(Integer.MAX_VALUE);
+    }
+
+    @Override
+    public synchronized void mark(final int readlimit) {
+        checkClosed();
+        markReadLimit = readlimit;
+        markPosition = currentPosition;
+    }
+
+    @Override
+    public synchronized void reset() {
+        checkClosed();
+        if (currentPosition == markPosition) {
+            return;
+        } else if (currentPosition > (markPosition + markReadLimit)) {
+            throw new MongoGridFSException("Mark readLimit has been exceeded, cannot reset");
+        }
+
+        eof = false;
```

Comment:
I'll clarify `'should mark and reset across chunks'` to explicitly go to eof and get -1 from a read.